In [1]:
import csv
import datetime

file = open('guns.csv')
data = list(csv.reader(file))

print(data[:5])


[['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education'], ['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4']]


In [2]:
headers = data[0]
data = data[1:len(data)]

print(data[:5])

[['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'], ['5', '2012', '02', 'Suicide', '0', 'M', '31', 'White', '100', 'Other specified', '2']]


In [3]:
years = [row[1] for row in data]
year_counts = {}

for year in years:
    if year in year_counts is not None:
        year_counts[year] += 1
    else:
        year_counts[year] = 1

print(year_counts)
        

{'2013': 33636, '2012': 33563, '2014': 33599}


In [4]:
dates = [datetime.datetime(year=int(row[1]), month=int(row[2]), day=1) for row in data]
print(dates[:5])
    

[datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0)]


In [5]:
date_counts = {}

for date in dates:
    if date in date_counts is not None:
        date_counts[date] += 1
    else:
        date_counts[date] = 0
print(date_counts)
        

{datetime.datetime(2012, 10, 1, 0, 0): 2732, datetime.datetime(2014, 2, 1, 0, 0): 2360, datetime.datetime(2013, 1, 1, 0, 0): 2863, datetime.datetime(2012, 12, 1, 0, 0): 2790, datetime.datetime(2012, 5, 1, 0, 0): 2998, datetime.datetime(2013, 3, 1, 0, 0): 2861, datetime.datetime(2014, 3, 1, 0, 0): 2683, datetime.datetime(2014, 7, 1, 0, 0): 2883, datetime.datetime(2013, 10, 1, 0, 0): 2807, datetime.datetime(2012, 4, 1, 0, 0): 2794, datetime.datetime(2012, 6, 1, 0, 0): 2825, datetime.datetime(2013, 5, 1, 0, 0): 2805, datetime.datetime(2012, 3, 1, 0, 0): 2742, datetime.datetime(2014, 12, 1, 0, 0): 2856, datetime.datetime(2013, 6, 1, 0, 0): 2919, datetime.datetime(2013, 11, 1, 0, 0): 2757, datetime.datetime(2013, 8, 1, 0, 0): 2858, datetime.datetime(2014, 1, 1, 0, 0): 2650, datetime.datetime(2014, 8, 1, 0, 0): 2969, datetime.datetime(2012, 1, 1, 0, 0): 2757, datetime.datetime(2012, 8, 1, 0, 0): 2953, datetime.datetime(2012, 7, 1, 0, 0): 3025, datetime.datetime(2012, 11, 1, 0, 0): 2728, date

In [6]:
sex_counts = {}
race_counts = {}
for row in data:
    sex = row[5]
    race = row[7]
    if sex in sex_counts:
        sex_counts[sex] += 1
    else:
        sex_counts[sex] = 1
    if race in race_counts:
        race_counts[race] += 1
    else:
        race_counts[race] = 1

def get_min_max(data):
    max_min = {}
    max_min["Max"] = max(data.keys(), key=(lambda k: data[k]))
    max_min["Min"] = min(data.keys(), key=(lambda k: data[k]))
    return max_min

min_max_sex = get_min_max(sex_counts)
min_max_race = get_min_max(race_counts)
min_max_dates = get_min_max(date_counts)
print(race_counts)
print(sex_counts)
print(min_max_sex)
print(min_max_race)
print(min_max_dates)
print(date_counts[min_max_dates["Max"]])
    

{'Black': 23296, 'Hispanic': 9022, 'White': 66237, 'Asian/Pacific Islander': 1326, 'Native American/Native Alaskan': 917}
{'M': 86349, 'F': 14449}
{'Max': 'M', 'Min': 'F'}
{'Max': 'White', 'Min': 'Native American/Native Alaskan'}
{'Max': datetime.datetime(2013, 7, 1, 0, 0), 'Min': datetime.datetime(2012, 2, 1, 0, 0)}
3078


## What we know so far
Based off the data that we have collected we can see that white males have the highest ammount of gundeaths. 

We can also see that Native American/Native Alaskan have the lowest ammount of gundeaths.

What this does not show is the precentage of the total populate for each classified race. We need to pull in the data from the United State Census and map our findings to equal pupulations distrabution. 

We will look at out findings based of a population of 100,000 for each race. 

In [7]:
file = open('census.csv')
census = list(csv.reader(file))
print(census)

[['Id', 'Year', 'Id', 'Sex', 'Id', 'Hispanic Origin', 'Id', 'Id2', 'Geography', 'Total', 'Race Alone - White', 'Race Alone - Hispanic', 'Race Alone - Black or African American', 'Race Alone - American Indian and Alaska Native', 'Race Alone - Asian', 'Race Alone - Native Hawaiian and Other Pacific Islander', 'Two or More Races'], ['cen42010', 'April 1, 2010 Census', 'totsex', 'Both Sexes', 'tothisp', 'Total', '0100000US', '', 'United States', '308745538', '197318956', '44618105', '40250635', '3739506', '15159516', '674625', '6984195']]


In [8]:
census_total = census[0][9]
census_data = census[1:len(census)][0]


mapping = {
    "Asian/Pacific Islander": census_data[14] + census_data[15],
    "Black": census_data[12],
    "Native American/Native Alaskan": census_data[13],
    "Hispanic": census_data[11],
    "White": census_data[10]
}

race_per_hundredk = {}

for race in race_counts:
    race_per_hundredk[race] = (int(race_counts[race]) / int(mapping[race])) * 100000

print(race_per_hundredk)


    




{'Black': 57.8773477735196, 'Hispanic': 20.220491210910907, 'White': 33.56849303419181, 'Asian/Pacific Islander': 8.746980714890115e-06, 'Native American/Native Alaskan': 24.521955573811088}


In [12]:
intents = [intent[3] for intent in data]
races = [race[7] for race in data]

homicide_race_counts = {}

for i, race in enumerate(races):
    if intents[i] == 'Homicide':
        if race in homicide_race_counts:
            homicide_race_counts[race] += 1
        else:
            homicide_race_counts[race] = 1
print(homicide_race_counts)

{'Black': 19510, 'Hispanic': 5634, 'White': 9147, 'Asian/Pacific Islander': 559, 'Native American/Native Alaskan': 326}


In [13]:
for race in homicide_race_counts:
    homicide_race_counts[race] = (int(homicide_race_counts[race]) / int(mapping[race])) * 100000

print(homicide_race_counts)

{'Black': 48.471284987180944, 'Hispanic': 12.627161104219914, 'White': 4.6356417981453335, 'Asian/Pacific Islander': 3.687452654316421e-06, 'Native American/Native Alaskan': 8.717729026240365}


## Findings

Looking the the Gun Deaths in the United States per 100,00 people for each race we can see the precntage of gun deaths as a result of homicide.

|Race | Gun deaths from Homicide |
|-----|:-----------------------:|
|Black|48.47%                   |
|Hispanic|12.63%
|Native ASmerican/Native Alaskan | 8.72%
|White | 4.64%
|Asian/Pacific Islander |3.69%
